libgadget/forcetree.c

```diff
@@ -106,74 +106,52 @@ static void force_validate_nextlist(const ForceTree * tree)
 }
 #endif
 
-static int
-force_tree_eh_slots_fork(EIBase * event, void * userdata)
-{
-    /* after a fork, we will attach the new particle to the force tree. */
-    EISlotsFork * ev = (EISlotsFork*) event;
-    int parent = ev->parent;
-    int child = ev->child;
-    ForceTree * tree = (ForceTree * ) userdata;
-    int no = force_get_father(parent, tree);
-    struct NODE * nop = &tree->Nodes[no];
-    /* FIXME: We lose particles if the node is full.
-     * At the moment this does not matter, because
-     * the only new particles are stars, which do not
-     * participate in the SPH tree walk.*/
-    if(nop->s.noccupied < NMAXCHILD) {
-       nop->s.suns[nop->s.noccupied] = child;
-       nop->s.Types += P[child].Type << (3*nop->s.noccupied);
-       nop->s.noccupied++;
-    }
-    if(child < tree->nfather && child >= 0)
-        tree->Father[child] = no;
-    return 0;
-}
-
 int
 force_tree_allocated(const ForceTree * tree)
 {
     return tree->tree_allocated_flag;
 }
 
 void
-force_tree_rebuild(ForceTree * tree, DomainDecomp * ddecomp, const int HybridNuGrav, const int DoMoments, const char * EmergencyOutputDir)
+force_tree_rebuild(ForceTree * tree, DomainDecomp * ddecomp, const ActiveParticles *act, const int HybridNuGrav, const int DoMoments, const char * EmergencyOutputDir)
 {
-    MPIU_Barrier(MPI_COMM_WORLD);
-    message(0, "Tree construction.  (presently allocated=%g MB)\n", mymalloc_usedbytes() / (1024.0 * 1024.0));
+    //message(0, "Tree construction.  (presently allocated=%g MB)\n", mymalloc_usedbytes() / (1024.0 * 1024.0));
 
     if(force_tree_allocated(tree)) {
         force_tree_free(tree);
     }
     walltime_measure("/Misc");
 
-    *tree = force_tree_build(PartManager->NumPart, ALLMASK, ddecomp, HybridNuGrav, DoMoments, EmergencyOutputDir);
+    *tree = force_tree_build(ALLMASK, ddecomp, act, HybridNuGrav, DoMoments, EmergencyOutputDir);
 
-    event_listen(&EventSlotsFork, force_tree_eh_slots_fork, tree);
```

Comment:
It was to add a newly created star particle to the tree. But now we rebuild the tree more often and in smaller chunks so it is no longer needed. Actually it was not needed before, because the only tree walk we did after star formation was the winds, which do not use star particles. Also it didn't really work reliably: it could only add particles to the lowest level of the tree.